### PR TITLE
Fix checkout service

### PR DESCRIPTION
**Pull request by grcevski**
# Changes

This PR updates the Go serivce (checkoutservice) to use the new 1.21 spec of the http OpenTelemetry APIs. It changes the go.mod file to use our grafana repos for `opentelemetry-go` and `opentelementry-go-contrib`. These forks of the opentelemetry repos have been updated to use: server.name, server.port, client.name, client.port instead of peer.name and peer.port, as well as change http.url to url.full.

The PR also fixes some other minor issues:
- The .NET service (cartservice) now runs with DOTNET_HOSTBUILDER__RELOADCONFIGONCHANGE to avoid this issue on Linux https://github.com/dotnet/AspNetCore.Docs/issues/19814
- The front-end memory for Kubernetes has been expanded to 500Mi, since 200Mi causes OOM.
- Ported docker compose commands for the Makefile.

## Merge Requirements

For new features contributions please make sure you have completed the following
essential items:

* [x ] `CHANGELOG.md` updated to document new feature additions
* [ x] Appropriate documentation updates in the [docs][]
* [ x] Appropriate Helm chart updates in the [helm-charts][]

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

[docs]: https://opentelemetry.io/docs/demo/
[helm-charts]: https://github.com/open-telemetry/opentelemetry-helm-charts

### CODE REVIEW

File: kubernetes/opentelemetry-demo.yaml [hunk 1]
```diff
@@ -7766,7 +7768,7 @@ spec:
             value: service.name=$(OTEL_SERVICE_NAME),service.instance.id=$(OTEL_K8S_POD_UID),service.namespace=opentelemetry-demo,k8s.namespace.name=$(OTEL_K8S_NAMESPACE),k8s.node.name=$(OTEL_K8S_NODE_NAME),k8s.pod.name=$(OTEL_K8S_POD_NAME)
           resources:
             limits:
-              memory: 200Mi
+              memory: 500Mi
```
Comment (kjrun316) [inline kubernetes/opentelemetry-demo.yaml]: Did we need 500Mi?  Just curious since I had it at 350Mi.

File: kubernetes/opentelemetry-demo.yaml [hunk 2]
```diff
@@ -7148,6 +7148,8 @@ spec:
             value: http://$(OTEL_COLLECTOR_NAME):4317
           - name: OTEL_RESOURCE_ATTRIBUTES
             value: service.name=$(OTEL_SERVICE_NAME),service.instance.id=$(OTEL_K8S_POD_UID),service.namespace=opentelemetry-demo,k8s.namespace.name=$(OTEL_K8S_NAMESPACE),k8s.node.name=$(OTEL_K8S_NODE_NAME),k8s.pod.name=$(OTEL_K8S_POD_NAME)
+          - name: DOTNET_HOSTBUILDER__RELOADCONFIGONCHANGE
```
Comment (kjrun316) [inline kubernetes/opentelemetry-demo.yaml]: Would it be helpful to add a comment or link to the issue?

https://github.com/dotnet/AspNetCore.Docs/issues/19814

Comment (grcevski) [reply kubernetes/opentelemetry-demo.yaml]: Done, thanks!

File: kubernetes/opentelemetry-demo.yaml (hunk 1 above)
Comment (grcevski) [reply kubernetes/opentelemetry-demo.yaml]: You are right, 350Mi is enough. I've updated it.

**Review by kjrun316 — APPROVED** [review]
Nice! Thanks!

**Comment by grcevski** [comment]
Thanks Karen!

**Pull request merged**